src/engine/engine_vfs.c

```diff
@@ -179,10 +179,10 @@ int mj_deleteFileVFS(mjVFS* vfs, const char* filename) {
       mju_free(vfs->filedata[i]);
 
       // scroll remaining files forward
-      while (i<vfs->nfile-1) {
-        mjSTRNCPY(vfs->filename[i], vfs->filename[i+1]);
-        vfs->filesize[i] = vfs->filesize[i+1];
-        vfs->filedata[i] = vfs->filedata[i+1];
+      for (int j = i; j < vfs->nfile - 1; ++j) {
```

Comment:
This loop should start at i+1 not i.

Reply:
I think `i` is correct. We delete the `i`th file and then always move e.g. `vfs->filesize[j] = vfs->filesize[j+1];`.

Reply:
You're right.